OpenSimRT/Common/include/CircularBuffer.h

```diff
@@ -45,9 +46,15 @@ template <int history, typename T> class CircularBuffer {
         current = 0;
         startOver = false;
         newValue = false;
+        externalNotification = false; // flag to remove the condition_variable
```

Comment:
In the docs of this class, we should summarize how this buffer works with examples. Show different scenarios of consumer-producer based on the flags. For example, is the buffer blocking the consumer thread and how the notification/synchronization works.